CRM/Contribute/BAO/Contribution.php

```diff
@@ -4072,24 +4075,23 @@ public static function updateMembershipBasedOnCompletionOfContribution($contribu
         unset($dates['end_date']);
         $membershipParams['status_id'] = $calcStatus['id'] ?? 'New';
       }
-      //we might be renewing membership,
-      //so make status override false.
+      // We might be renewing membership so make status override false.
       $membershipParams['is_override'] = FALSE;
       $membershipParams['status_override_end_date'] = 'null';
-      $membership = civicrm_api3('Membership', 'create', $membershipParams);
-      $membership = $membership['values'][$membership['id']];
+      $membershipBAO = CRM_Member_BAO_Membership::create($membershipParams);
```

Comment:
Can we use `writeRecord()` or apiv4 - I'm trying NOT to call the `create()` functions directly

Reply:
Ideally yes... but writeRecord() bypasses *all* the logic in create which would be nice but requires a lot more rework. Calling via api3 forces the dreaded `$ids['membership']` param to be passed in to create so I thought bypassing api3 initially would allow us to resolve that.
But depending on how much testing we have around this maybe we could actually unwrap all that and go to API4/writeRecord()

Reply:
what about apiv4?

Reply:
I've moved this part to https://github.com/civicrm/civicrm-core/pull/32257 as this PR is now all about extracting MembershipPayment usage out.